lib/_emerge/depgraph.py

```diff
@@ -5018,6 +5018,21 @@ def _resolve(self, myfavorites):
                     pkg, existing_node = self._select_package(
                         myroot, atom, onlydeps=onlydeps
                     )
+
+                    # Is the package installed (at any version)?
+                    if pkg:
+                        package_is_installed = False
+                        if "update_if_installed" in self._dynamic_config.myparams:
+                            for any_match in self._iter_match_pkgs_any(
```

Comment:
We can used the `_iter_match_pkgs` method here with pkg_type "installed".

Reply:
oh, nice!